website/docs/guides/customize-schema-alias.md

```diff
@@ -0,0 +1,473 @@
+|-------------|:-------------:|:-------------:|:------------------------------:|
+| Production  | prod          | analytics     |prod.marketing.my_model         |
+| Developer 1 | dev           | dbt_dev1      |dev.marketing.my_model          |
+| Developer 2 | dev           | dbt_dev2      |dev.marketing.my_model          |
+| CI PR 123   | ci            | dbt_pr_123    |ci.marketing.my_model           |
+| CI PR 234   | ci            | dbt_pr_234    |ci.marketing.my_model           |
+
+
+We can see that both Developer 1 and Developer 2 get the same object for `my_model`. This means that if they both work on this model at the same time, it will be impossible to know if the version currently in the data warehouse is the one from Developer 1 and Developer 2.
+
+Similarly, different PRs will result in the exact same object in the data warehouse. If different PRs are open at the same time and modifying the same models, it is very likely that we will get issues, slowing down the whole development and code promotion.
+
+
+### Solution
+
+As described above in the first example, the macro should be updated to check if we are currently running dbt in Production or not. It is only when in Production that we want to remove the concatenation and only use the custom schema.
+
+
+## More tips and tricks
+
+
+### Creating non existing databases from dbt
+
+dbt will automatically try to create a schema if it doesn’t exist and if an object needs to be created in it, but it won’t automatically try to create a database that doesn’t exist.
+
+So, if your `generate_database_name()` configuration points to different databases, which might not exist, dbt will fail if you do a simple `dbt build`. 
+
+It is still possible to get it working in dbt by creating some macros that will check if a database exists and if not, dbt will create it. You can then call those macros either in [a `dbt run-operation ...` step](https://docs.getdbt.com/reference/commands/run-operation) in your jobs or as a [`on-run-start` hook](/reference/project-configs/on-run-start-on-run-end).
+
+
+### Assuming context using Environment Variables rather than `target.name`
+
+
+We prefer to use [environment variables](/docs/build/environment-variables) over `target.name` ([documentation here](/reference/dbt-jinja-functions/target)) to decipher the context of the dbt invocation. 
+
+- [`target.name`](http://target.name) cannot be set at the environment-level. Therefore, every job within the environment must explicitly specify the `target.name` override. If the job does not have the appropriate `target.name` value set, the database/schema/alias may not resolve properly. Alternatively, environment variable values are inherited by the jobs within their corresponding environment. The environment variable values can also be overwritten within the jobs if needed.
+
+
+<Lightbox src="/img/docs/dbt-cloud/using-dbt-cloud/Environment-Variables/custom-schema-env-var-targetname.png." title="ADD TITLE" />
```

Comment:
the images aren't rendering for me. for _all_ the images in this docs, i suggest:
- updating the image links
- adding a title/image caption

<img width="821" alt="Screenshot 2024-07-08 at 10 43 22" src="https://github.com/dbt-labs/docs.getdbt.com/assets/89008547/2cea99d7-f063-4ddf-8155-5a38d82cce7a">